iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -260,6 +260,20 @@ def has_gpr_offsets(node: fx.Node) -> bool:
                     gpr_offset_dim
                 ] = updated_dim_with_gpr_offset
 
+                if hasattr(custom, "mapping_dynamic_vals"):
+                    new_dynamic_vals = []
+                    for dyn_val in custom.mapping_dynamic_vals:
+                        if any(
+                            sympy.sympify(v.start).has(GPR_NUM)
+                            for v in get_custom(dyn_val).index.values()
+                        ):
+                            extract = ExtractSlice(
```

Comment:
Just curious, do we need to actually extract_slice on the dyn_vals? what example would it fail on if we just pass in dynamic vals directly without extract?

Reply:
`self_index` with `GPR_NUM` gets partitioned 4x4 ops which are then reshaped into 16 vec. Read op which is using it is also partitioned into 4x4, so we need to extract 4-sized chunks back.

Reply:
the ReadOp being the Read in charge of reading the rpe bias?

Reply:
Can we add some comments just explaining this on the code actually? :)

Reply:
added some comments

Reply:
thanks!